pkg/build/controller/build/build_controller.go

```diff
@@ -366,12 +374,13 @@ func (bc *BuildController) createPodSpec(originalBuild *buildapi.Build, ref stri
 
 // resolveOutputDockerImageReference returns a reference to a Docker image
 // computed from the buid.Spec.Output.To reference.
-func (bc *BuildController) resolveOutputDockerImageReference(build *buildapi.Build) (string, error) {
+func (bc *BuildController) resolveOutputDockerImageReference(build *buildapi.Build) (string, *kapi.LocalObjectReference, error) {
```

Comment:
Similarly, why are we returning the secret here ... a comment/context would be useful.  Currently the comment only addresses the reference returned.